source/frontend/src/features/mapSideBar/property/tabs/propertyDetails/update/models/UpdatePropertyDetailsFormModel.ts

```diff
@@ -129,99 +133,116 @@ export class UpdatePropertyDetailsFormModel {
-      isRwyBeltDomPatent: this.isRwyBeltDomPatent,
-      latitude: this.latitude,
-      longitude: this.longitude,
-      location: { coordinate: { x: this.longitude, y: this.latitude } },
-      landLegalDescription: this.landLegalDescription,
-      landArea: this.landArea,
-      areaUnit: toTypeCode(this.areaUnitTypeCode),
+      id: this.id ?? 0,
+      rowVersion: this.rowVersion ?? null,
+      pid: this.pid ?? null,
+      pin: this.pin ?? null,
+      planNumber: this.planNumber ?? null,
+      zoning: this.zoning ?? null,
+      zoningPotential: this.zoningPotential ?? null,
+      municipalZoning: this.municipalZoning ?? null,
+      notes: this.notes ?? null,
+      name: this.name ?? null,
+      description: this.description ?? null,
+      isSensitive: this.isSensitive ?? false,
+      pphStatusTypeCode: this.pphStatusTypeCode ?? null,
+      isRwyBeltDomPatent: this.isRwyBeltDomPatent ?? null,
+      latitude: this.latitude ?? null,
+      longitude: this.longitude ?? null,
+      location: { coordinate: { x: this.longitude ?? 0, y: this.latitude ?? 0 } },
+      landLegalDescription: this.landLegalDescription ?? null,
+      landArea: this.landArea ?? null,
+      areaUnit: toTypeCodeNullable(this.areaUnitTypeCode),
       isVolumetricParcel: stringToBoolean(this.isVolumetricParcel as string),
-      volumetricMeasurement: this.volumetricMeasurement,
-      volumetricUnit: toTypeCode(this.volumetricUnitTypeCode),
-      volumetricType: toTypeCode(this.volumetricParcelTypeCode),
-      propertyType: toTypeCode(this.propertyTypeCode),
-      status: toTypeCode(this.statusTypeCode),
-      district: toTypeCode(this.districtTypeCode),
-      region: toTypeCode(this.regionTypeCode),
-      address: this.address !== undefined ? this.address.toApi() : undefined,
-      generalLocation: stringToUndefined(this.generalLocation),
+      volumetricMeasurement: this.volumetricMeasurement ?? null,
+      volumetricUnit: toTypeCodeNullable(this.volumetricUnitTypeCode),
+      volumetricType: toTypeCodeNullable(this.volumetricParcelTypeCode),
+      propertyType: toTypeCodeNullable(this.propertyTypeCode),
+      status: toTypeCodeNullable(this.statusTypeCode),
+      district: toTypeCodeNullable(this.districtTypeCode),
+      region: toTypeCodeNullable(this.regionTypeCode),
+      address: exists(this.address) ? this.address.toApi() : null,
+      generalLocation: stringToNull(this.generalLocation),
       // multi-selects
-      anomalies: this.anomalies?.map(e => e.toApi()),
-      tenures: this.tenures?.map(e => e.toApi()),
-      roadTypes: this.roadTypes?.map(e => e.toApi()),
-      adjacentLands: this.adjacentLands?.map(e => e.toApi()),
+      anomalies: this.anomalies?.map(e => e.toApi()) ?? null,
+      tenures: this.tenures?.map(e => e.toApi()) ?? null,
+      roadTypes: this.roadTypes?.map(e => e.toApi()) ?? null,
+
+      boundary: null,
+      dataSource: null,
+      dataSourceEffectiveDateOnly: EpochISODateTimeString,
```

Comment:
Hmm, I guess this is the equivalent of setting null for two required fields? would we be better storing and passing of the actual value if we have it?

Reply:
that would be the ideal. in this case I chose to preserve the existing functionality as much as posible